src/main/scala/fi/aalto/cs/apluscourses/intellij/utils/ModuleUtils.scala

```diff
@@ -34,4 +36,66 @@ object ModuleUtils {
     nonEmpty
   }
 
+  def naiveValidate(@NotNull command: String): Boolean =
+    command.matches("import\\so1\\.[a-z]*(\\_|\\._)$")
+
+  def clearCommands(@NotNull imports: Array[String]): Array[String] =
+    imports
+      .clone
+      .map(_.replace("import ", ""))
+      .map(_.replace("._", ""))
+
+  def getCommandsText(@NotNull imports: Array[String]): String =
+    imports.length match {
+      case 0 => ""
+      case 1 => "Auto-imported package [" + imports(0) + "] for your convenience." // scalastyle:ignore
+      case _ => "Auto-imported packages [" + imports.mkString(", ") + "] for your convenience."
+    }
+
+  def getUpdatedText(@NotNull module: Module,
+                     @NotNull commands: Array[String],
+                     @NotNull originalText: String): String = {
+    val commonText = "Write a line (or more) of " +
+      "Scala and press [Ctrl+Enter] to run it. Use [Up] and [Down] to scroll through your earlier " +
+      "inputs. \nChanges to the module are not loaded automatically. If you edit the files, restart" +
+      " the REPL with [Ctrl+F5] or the icon on the left. \n"
+
+    if (isTopLevelModule(module)) {
+      commonText + originalText + "Note: This REPL session is not linked to any course module. To " +
+        "use a module from the REPL, select the module and press [Ctrl+Shift+D] to launch a new " +
+        "session."
+    } else {
+      val validCommands = commands.filter(command => naiveValidate(command))
+      val clearedCommands = clearCommands(validCommands)
+      val commandsText = getCommandsText(clearedCommands)
+
+      "Loaded A+ Courses module [" + module.getName + "]. " + commandsText + "\n" + commonText +
+        originalText
+    }
+  }
+
+  def getModuleRoot(@NotNull moduleFilePath: String): String = {
+    val lastIndexOf = moduleFilePath.lastIndexOf("/")
+    moduleFilePath.substring(0, lastIndexOf + 1) // scalastyle:ignore
+  }
+
+  def initialReplCommandsFileExist(@NotNull replCommandsFileName: String,
+                                   @NotNull moduleFilePath: String): Boolean = {
+    val moduleDir = getModuleRoot(moduleFilePath)
+    Files.exists(Path.of(moduleDir + replCommandsFileName))
+  }
+
+  @NotNull
```

Comment:
@nikke234 I tried to call this guy from Java, but failed... is there some easy way?

Reply:
Weird, I'm struggling to get it to work either

Reply:
May be relevant: https://stackoverflow.com/questions/23261075/compiling-scala-before-alongside-java-with-gradle